advisor/src/main/kotlin/advisors/VulnerableCode.kt

```diff
@@ -144,7 +145,12 @@ class VulnerableCode(
             vulnerabilityUrls.map { extractVulnerabilityId(it) }
                 .map { async { service.getVulnerability(it) } }
                 .awaitAll()
-                .map { Vulnerability(it.cveId.orEmpty(), it.cvss ?: SEVERITY_UNDEFINED, URI(it.url)) }
-                .associateBy { "${it.url}" }
+                .map {
+                    Vulnerability(
+                        it.cveId.orEmpty(),
+                        listOf(VulnerabilityReference(URI(it.url), null, (it.cvss ?: SEVERITY_UNDEFINED).toString()))
+                    )
+                }
+                .associateBy { "${it.references.first().url}" }
```

Comment:
Nit: While at it, consider using `it.references.first().url.toString()` instead.